lib/src/list/list_builder.dart

```diff
@@ -15,6 +15,9 @@ class ListBuilder<E> {
   late List<E> _list;
   _BuiltList<E>? _listOwner;
 
+  /// Instantiates with elements from an [Iterable<E>].
+  factory ListBuilder.of(Iterable<E> elements) => ListBuilder<E>(elements);
```

Comment:
This ends up calling `replace` which doesn't benefit from the collection type being known, how about adding a `_replaceOf` method which uses `List.of` instead of `List.from`, then calling that here.

Private because the end goal is that "replace" turns into "replaceOf", but we can worry about that later :)